PyInstaller/hooks/hook-shapely.py

```diff
@@ -0,0 +1,28 @@
+import os
+
+from PyInstaller.utils.hooks import get_package_paths
+from PyInstaller import compat
+
+# Necessary when using the vectorized subpackage
+hiddenimports = ['shapely.prepared']
+
+pkg_base, pkg_dir = get_package_paths('shapely')
+
+if compat.is_win:
+    binaries = []
+    lib_dir = os.path.join(pkg_dir, 'DLLs')
+    binaries += [(os.path.join(lib_dir, f), '') for f in os.listdir(lib_dir)]
+elif compat.is_linux:
+    binaries = []
+    lib_dir = os.path.join(pkg_dir, '.libs')
+    dest_dir = os.path.join('shapely', '.libs')
+
+    # This duplicates the libgeos*.so* files in the build.  PyInstaller will
+    # copy them into the root of the build by default, but shapely cannot load
+    # them from there in linux IF shapely was installed via a whl file. The
+    # whl bundles its' own libgeos with a different name, something like
+    # libgeos_c-*.so.* but shapely tries to load libgeos_c.so if there isn't a
+    # ./libs directory under its' package. There is a proposed fix for this in
+    # shapely but it has not been accepted it:
+    # https://github.com/Toblerity/Shapely/pull/485
+    binaries += [(os.path.join(lib_dir, f), dest_dir) for f in os.listdir(lib_dir)]
```

Comment:
Toblerity/Shapely#485 is merged now. You could make this depending on the version by using `PyInstaller.utils.hooks.is_module_satisfies()`, see e.g. PyInstaller/hooks/hook-botocore.py.